web/libs/datamanager/src/components/CellViews/Agreement/Agreement.jsx

```diff
@@ -24,20 +28,31 @@ const formatNumber = (num) => {
 export const Agreement = (cell) => {
   const { value, original: task } = cell;
   const sdk = useSDK();
-  const agreementCN = cn("agreement");
-  const scoreElem = agreementCN.elem("score");
-  const handleClick = useCallback(
-    (e) => {
-      sdk.invoke("agreementCellClick", e, task);
-    },
-    [sdk, task],
+  const [content, setContent] = useState(null);
+  const isAgreementPopoverEnabled =
+    window.APP_SETTINGS.billing?.enterprise && ff.isActive(FF_AVERAGE_AGREEMENT_SCORE_POPOVER);
+
+  const handleClick = isAgreementPopoverEnabled
+    ? (e) => {
+        e.preventDefault();
+        e.stopPropagation();
+        sdk.invoke("agreementCellClick", { task }, (jsx) => setContent(jsx));
+      }
+    : undefined;
+
+  const score = (
+    <span className={agreementScoreTextColor(value)}>{isDefined(value) ? `${formatNumber(value)}%` : ""}</span>
   );
 
   return (
-    <div className={agreementCN.toString()} onClick={handleClick}>
-      <span className={clsx(scoreElem.toString(), scoreElem.mod({ [agreement(value)]: true }).toString())}>
-        {isDefined(value) ? `${formatNumber(value)}%` : ""}
-      </span>
+    <div className="flex items-center" onClick={handleClick}>
+      {isAgreementPopoverEnabled ? (
+        <Popover trigger={score} align="start">
```

Comment:
so what's the idea here? is it triggered by clicking on `score` but the `content` is empty, so it will be in a confusing state? can we move content loading closer to Popover (does it have any events?) and add a loading state?

Reply:
I see that all this logic is inside `agreementCellClick` handler in LSE... Given all this interconnection that looks like a good option to trigger it, I see...